backend/ethereum/channel/adjudicator.go

```diff
@@ -112,7 +113,7 @@ type adjFunc = func(
 
 // call calls the given contract function `fn` with the data from `req`.
 // `fn` should be a method of `a.contract`, like `a.contract.Register`.
-func (a *Adjudicator) call(ctx context.Context, req channel.AdjudicatorReq, fn adjFunc) error {
+func (a *Adjudicator) call(ctx context.Context, req channel.AdjudicatorReq, fn adjFunc, fnName string) error {
```

Comment:
I think you should document the new parameter.

I don't love this manual way of providing the function name. I think it's not super reliable, because we may end up calling the same function under different names, if we are not careful. A more reliable way would be to introduce an enum type that ensures that you can only call certain functions with their name attached. Anyhow, it's not super important, but I wanted to point that out.

Reply:
Yeah, I agree that we should document this parameter.

Also, using enum for func names would be more reliable. Thanks for pointing it out. I will update the PR with the required change.

Reply:
I have updated the code to use enum definitions.

Reply:
In the [first update](https://github.com/hyperledger-labs/go-perun/compare/edda92c04127674699991b5c9ceb4d7ab8922103..d7cc92abeb076bc45ef9289acd13662ed54aefd5), I have added the enum in the core [client] package. But the it appeared to me like breaking the abstraction of the core package, because the specific on-chain transactions may depend upon the contract implementations for the blockchain backend.

So in the [following update](https://github.com/hyperledger-labs/go-perun/compare/d7cc92abeb076bc45ef9289acd13662ed54aefd5..04b1bd0bd5714299c1cff39186a6b5bf9e8855ee), I have moved the enum definitions to [backend/ethereum/channel] package. As i felt it belongs better here.

What are your thoughts on this ? @matthiasgeihs  @ggwpez